### PR TITLE
feature/447-add-dark-theme

### LINKED ISSUE #447: Add dark theme
# User Story

### As a: 
User

### I want:
to choose between the current light them and a dark theme

### so that:
my eyes don't hurt that much after using the platform (at night)

## Acceptance criteria:
- [ ] let the user choose which theme he wants
- [ ] design a dark theme

## Additional info:
- we could make the change time depending, so eg after 10 o'clock pm till 6 o'clock the dark theme applies automatically if wanted

------
_Please tag this issue if you are sure to which tag(s) it belongs._

### PR DESCRIPTION
**Pull request by AndreasSchroll**
## Description:
add dark theme
user can choose theme (default, night, auto)
in automode the theme becomes dark between 10PM and 6AM

Closes #447 

## Improvements
- 

## Known Issues:
_NONE_

------
_Remember to prefix your PR-Title with `⚠ WIP: ` if you still work on it.
If you have reached a final state, remove the prefix._

### CODE REVIEW

File: app/webFrontend/src/app/shared/services/theme.service.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+export class ThemeService {
+  private renderer: Renderer2;
+  themes: string[]= ['default', 'night', 'auto'];
+
+  constructor(private rendererFactory: RendererFactory2) {
+    this.renderer = rendererFactory.createRenderer(null, null);
+  }
+
+  setTheme(theme: string) {
+    this.renderer.removeClass(document.body, 'nightTheme');
```
Comment (aeimer) [inline app/webFrontend/src/app/shared/services/theme.service.ts]: Wouldn't it be good to run over all possible theme-classes and remove them?

Furthermore is it possible to remove them by wildcard?
`removeClass(document.body, 'theme-*')`
then all theme-classes would need to be named like `theme-dark, theme-default, theme-contrast, ...`

I also would like to see these CONSTANT names of the themes in at least Constants or an Enum-class ;)

File: app/webFrontend/src/app/user/user-edit/user-edit.component.ts [hunk 2]
```diff
@@ -60,7 +64,8 @@ export class UserEditComponent implements OnInit {
       this.userForm.patchValue({
         profile: {
           firstName: this.user.profile.firstName,
-          lastName: this.user.profile.lastName
+          lastName: this.user.profile.lastName,
+          theme: (this.themes.indexOf(this.user.profile.theme) !== -1 ? this.user.profile.theme : 'default'),
```
Comment (aeimer) [inline app/webFrontend/src/app/user/user-edit/user-edit.component.ts]: Is this the right way to read a dropdown box?

File: app/webFrontend/src/_variables.scss [hunk 3]
```diff
@@ -29,4 +29,9 @@ $bp-width--xl: 75em; // 1200px
 $bp-width--xxl: 121em; // 1936px
 $bp-width--max: 160em; // 2560px => app-root max-width
 
+$app-night-primary: mat-palette($mat-indigo);
+$app-night-accent: mat-palette($mat-pink, A200, A100, A400);
+// The warn palette is optional (defaults to red).
+$app-night-warn: mat-palette($mat-red);
+
```
Comment (aeimer) [inline app/webFrontend/src/_variables.scss]: Please move these lines up, so all the color/theme vars are positioned in one block

File: app/webFrontend/src/app/user/user-edit/user-edit.component.ts (hunk 2 above)
Comment (AndreasSchroll) [reply app/webFrontend/src/app/user/user-edit/user-edit.component.ts]: There, the selected one is set

Comment (aeimer) [reply app/webFrontend/src/app/user/user-edit/user-edit.component.ts]: ah, I got it

**Comment by aeimer** [comment]
# Findings
## Course edit
### Course edit - Content
The black text and the black dots need to be white or something
grafik

### Course edit - Member
#### Course edit - Member - on Hover
The on hover needs a nother color
grafik

#### Course edit - Member - search
The serach needs a better background color
grafik

### Course edit - General
Select field needs to get better color. This may be a general problem?!
grafik

**Comment by aeimer** [comment]
The member stuff is better, but on drag and drop:
grafik

**Comment by aeimer** [comment]
@AndreasSchroll are we ready to merge?

**Comment by AndreasSchroll** [comment]
Yes

**Comment by aeimer** [comment]
Tested locally, works. Will merge.

**Pull request merged**